src/core/REPL.pm6

```diff
@@ -402,9 +402,18 @@ do {
         method history-file(--> Str:D) {
             return $!history-file.absolute if $!history-file.defined;
 
-            $!history-file = $*ENV<RAKUDO_HIST>
-                ?? $*ENV<RAKUDO_HIST>.IO
-                !! ($*HOME || $*TMPDIR).add('.perl6/rakudo-history');
+            with $*ENV<RAKUDO_HIST> {
+                $!history-file = $_.IO;
+            }
+            else {
+                # looking up the home thingy from the CompUnit Repository Registry would be more elegant
+                my IO::Path $home = "$*HOME/.perl6".IO;
```

Comment:
Without commenting on the rest of the PR...  Note that `$*HOME` can be undefined, in which case you'll be attempting to create `.perl6` in the root dir:

```
2018.03.129 zoffix@VirtualBox~$ unset HOME
2018.03.129 zoffix@VirtualBox/home/zoffix$ perl6 -e 'say "$*HOME/.perl6".IO'                                                                                                                                                                                   
Use of Nil in string context
  in block <unit> at -e line 1
"/.perl6".IO
```